src/app/tests/TestEventOverflow.cpp

```diff
@@ -52,9 +51,23 @@ static uint8_t gInfoEventBuffer[2048];
 static uint8_t gCritEventBuffer[2048];
 static chip::app::CircularEventBuffer gCircularEventBuffer[3];
 
-class TestContext : public chip::Test::AppContext
+class TestEventOverflow : public ::testing::Test
 {
 public:
+    static void SetUpTestSuite()
+    {
+        mpTestContext = new chip::Test::AppContext;
+        mpTestContext->SetUpTestSuite();
+    }
+    static void TearDownTestSuite()
+    {
+        mpTestContext->TearDownTestSuite();
+        if (mpTestContext != nullptr)
```

Comment:
same comment here: mpTestContext is used already, so check seems redundant. I assume this may be throughout the PR. 

I wonder if given this is test code, maybe we can assume that "new" does not really fail so we can skip all these nullability checks. We have to trust the framework that TearDown is called only after SetUp.

Reply:
maybe I have trust issues....done